sim/src/main/makefrag/firesim/build.mk

```diff
@@ -2,7 +2,7 @@
 
 $(FIRRTL_FILE) $(ANNO_FILE): $(TARGET_CP)
 	@mkdir -p $(@D)
-	java -cp $$(cat $(TARGET_CP)) chipyard.Generator \
+	cd $(base_dir) && java -cp $$(cat $(TARGET_CP)) chipyard.Generator \
```

Comment:
What exactly breaks if the working directory is not changed?

Reply:
https://github.com/ucb-bar/chipyard/blob/95f30d0411645b12dfc5c0b14f9af0a5a678a991/generators/firechip/src/main/scala/TargetConfigs.scala#L29-L30 selects the BootROM based off the working directory. In the FireSim-as-Top case, the working directory is `firesim/sim` and the `target-rtl/chipyard` directory is checked out (submodule is init'ed) allowing the code to find the right bootrom. However in the Chipyard-as-Top case (without the fix) things are a bit different. The working directory is still `firesim/sim`, so the code linked 1st tries to find the BootROM in `<working-dir>/generators/...` (a CY path) and fails to find it there. Next it looks in `<working-dir>/target-rtl/chipyard/...`. However, this also fails since the `target-rtl/chipyard` submodule is uninitialized. Changing the working directory to Chipyard case avoids this.